### ISSUE TITLE
json_encode error when serializing closure that uses invalid UTF-8 characters

### ISSUE THREAD

**Issue by tomlankhorst** [opis/closure#34]
The following test reproduces the issue.

```php
public function testSecureInvalidUtf8Data()
{
    $a = utf8_decode("Düsseldorf"); // this returns a deliberately malformed string
    $closure = function() use($a)
    {
        return $a;
    };

    SerializableClosure::setSecretKey('secret');

    $value = serialize(new SerializableClosure($closure));
    $u = unserialize($value);

    $this->assertEquals($a, $u->getClosure()());
}
```

This causes the closure serialization to fail:

```
Opis\Closure\SecurityException : Your serialized closure might have been modified and it's unsafe to be unserialized. Make sure you use the same security provider, with the same settings, both for serialization and unserialization.
```

Use-cases for this are a closure that depends on external data, which may be incorrectly formatted, but that shouldn't withhold the closure from serializing imo.

**Comment by tomlankhorst** [comment]
Thank you!

**Comment by msarca** [comment]
This is now fixed in [v3.3.0](/opis/closure/releases/tag/3.3.0). Thanks for reporting this bug.
